src/test.js

```diff
@@ -677,8 +709,10 @@ export function internalStop( test ) {
 
 // Forcefully release all processing holds.
 function internalRecover( test ) {
-	test.semaphore = 0;
-	internalStart( test );
+	if ( config.blocking ) {
```

Comment:
My assumption is this should always hold (or rather, if `config.blocking` is falsy, then there's really nothing to do with regard to starting the test). If this is a bad assumption, let me know.